go.mod

```diff
@@ -4,10 +4,14 @@ go 1.16
 
 require (
 	github.com/MakeNowJust/heredoc v1.0.0
+	github.com/Microsoft/go-winio v0.4.17 // indirect
 	github.com/blang/semver v3.5.1+incompatible
+	github.com/containerd/containerd v1.4.4 // indirect
 	github.com/coredns/corefile-migration v1.0.11
 	github.com/davecgh/go-spew v1.1.1
 	github.com/docker/distribution v2.7.1+incompatible
+	github.com/docker/docker v20.10.6+incompatible
```

Comment:
A little concerned here to import docker as a library directly in our primary go.mod, any alternatives?

Reply:
@vincepri would it be fine if we add a new go module as suggested by @sbueringer?
The main benefit I can see in this work is that it will allow us to get rid of the docker CLI embedded in CAPD providers, so the image will shrink in size, became faster to build and download, and most important we can finally move CAPD to images to gcr.io/distroless/static:nonroot

Reply:
Having a test framework sub-module requires folks to use a commit on that submodule instead of the tag, or we need to revamp our release process to also tag the test framework.

As an alternative, have we thought about moving the test framework into a different repository?

Reply:
I'm not that familiar with go modules (apparently). Can only the "main" go module be referenced by the git tag? What would we have to do to tag a go submodule? We currently already have separate Go modules for hack/tools and CAPD. 
 At least hack/tools is also imported by other providers (but probably always by commit).

I'm a bit concerned about moving the test/framework in a separate repo at this point. I think this will add a lot of overhead to the development of e2e tests. Right now it's a huge benefit to have the core controllers, kubeadm providers, CAPD *and* the test framework all in one repo.